presto-cache/src/test/java/com/facebook/presto/cache/alluxio/TestAlluxioCachingFileSystem.java

```diff
@@ -93,11 +87,47 @@ public void setupMethod()
     @Test(timeOut = 30_000)
     public void testBasic()
             throws Exception
+    {
+        testWithDataValidation(false);
+    }
+
+    @Test(timeOut = 30_000)
+    public void testBasicWithDataValidationEnabled()
+            throws Exception
+    {
+        testWithDataValidation(true);
+    }
+
+    @Test(invocationCount = 10)
+    public void testStress()
+            throws ExecutionException, InterruptedException, URISyntaxException, IOException
     {
         CacheConfig cacheConfig = new CacheConfig()
                 .setCacheType(ALLUXIO)
                 .setCachingEnabled(true)
                 .setBaseDirectory(cacheDirectory);
+        AlluxioCacheConfig alluxioCacheConfig = new AlluxioCacheConfig()
+                .setMaxCacheSize(new DataSize(10, KILOBYTE));
+
+        AlluxioCachingFileSystem cachingFileSystem = cachingFileSystem(cacheConfig, alluxioCacheConfig);
+        stressTest(data, (position, buffer, offset, length) -> {
+            try {
+                readFully(cachingFileSystem, position, buffer, offset, length);
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    private void testWithDataValidation(boolean validationEnabled)
```

Comment:
Can we keep the base method name as testBasic and use it to leverage validation on/off?